### DISCUSSION TITLE
Smartcard Support

### DISCUSSION THREAD

**Discussion by kaaax0815** [the-djmaze/snappymail#1553] [Ideas]
I currently have my gpg keys on a yubikey.

Would be nice if snappymail would support using it

**Comment by the-djmaze** [comment]
That will not work as the browser or server can't read the yubikey, only act on FIDO2

**Reply by kaaax0815** [reply]
There is a Web USB API but I don't know if it can be used for that

**Reply by the-djmaze** [reply]
But then still: how to access the keys and use them?

**Reply by kaaax0815** [reply]
You can't access the keys. But I would be nice to be able to sign/decrypt emails using snappymail using yubikey without having to import the secret key. As the secret key live securely on the security key

**Reply by the-djmaze** [reply]
For that gnupg must interact with the yubikey.
So snappymail must be running on the same computer where the yubikey is installed and PHP must have access to gnupg.
This is not something that snappymail is designed for.

**Reply by kaaax0815** [reply]
It's seems there were already people thinking of the idea
https://github.com/openpgpjs/openpgpjs/issues/1111
https://github.com/openpgpjs/openpgpjs/issues/362
https://github.com/FlowCrypt/flowcrypt-browser/issues/964
https://github.com/openpgpjs/openpgpjs/issues/553

But it seems like this isn't possible currently. But may be in the feature

**Reply by kaaax0815** [reply]
https://github.com/mailvelope/gpgmejs
This communicates with the local gpg agent. So snappymail would only need to integrate with that

Edit: this requires creating a extension. Mailvelope uses this

**Reply by the-djmaze** [reply]
Correct. Like i said it is currently hardly to not possible.
